### PR TITLE
Elements: Deprecate old block support filter callbacks

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Addresses: https://github.com/WordPress/gutenberg/pull/59535#issuecomment-1975689098

## What?

- Deprecates remove block support filter callback function
- Deprecates argument for callback function that was moved to a different filter
- Fixes some code style issues

## Why?

Whether the callback functions were part of a private API and not documented publicly or not, if it could be used, it might have been so better to deprecate the function or argument as required.

## How?

- Deprecates `gutenberg_render_elements_support`
- Deprecates string value for first argument for `gutenberg_render_elements_support_styles` (now it is used with `render_block_data` the arg should be an array).
- Fixes some code style issues

## Testing Instructions

1. Run unit tests: `npm run test:unit:php:base -- --filter WP_Block_Supports_Elements_Test`
2. Test element styles can still be applied and rendered correctly on the frontend following the steps over on https://github.com/WordPress/gutenberg/pull/59535.

## Next Steps

- [x] Iterate on this alongside the corresponding backport of https://github.com/WordPress/gutenberg/pull/59535

### CODE REVIEW

**Comment by aaronrobertshaw** [comment]
I'll be AFK for just over a week, so my connection and bandwidth might be limited in addressing any feedback until the middle of next week.

**Review by andrewserong — APPROVED** [review]
Thanks for the follow-up! The elements styles are still being output correctly, and the deprecated calls all look good to me. I'm not super confident with my understanding of PHP deprecated calls, though, so it might be worth getting a second opinion from @peterwilsoncc, but overall this LGTM!

**Comment by peterwilsoncc** [comment]
Acknowledging I've seen this but I'm ignoring anything that isn't going to hit WordPress 6.5 until after the release.

File: lib/block-supports/elements.php
Comment (peterwilsoncc) [inline lib/block-supports/elements.php]: This refers to itself as the replacement in the third param.

**Review by peterwilsoncc — COMMENTED** [review]
This looks good to me folks, just a minor note about a self referential deprecation notice.

Once that's fixed, this is good to merge.

File: lib/block-supports/elements.php
Comment (aaronrobertshaw) [reply lib/block-supports/elements.php]: Thanks for catching that @peterwilsoncc! 🙈 

I've fixed it and the docblock in https://github.com/WordPress/gutenberg/pull/59538/commits/92ebee547c73bb5f30087787f6e39f379f1a6603

**Comment by aaronrobertshaw** [comment]
The changes from this PR and https://github.com/WordPress/gutenberg/pull/59535 are combined in a single backport available in https://github.com/WordPress/wordpress-develop/pull/6214.

**Pull request merged**
